homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -380,82 +393,86 @@ def _build_observations_by_template(self):
 
     def _process_state(self, entity_observation):
-        """Return True if state conditions are met."""
-        entity = entity_observation["entity_id"]
+        """Return True if state conditions are met, return False if they are not.
+
+        Returns None if the state is unavailable.
+        """
+
+        entity = entity_observation.entity_id
 
         try:
             if condition.state(self.hass, entity, [STATE_UNKNOWN, STATE_UNAVAILABLE]):
                 return None
 
-            return condition.state(
-                self.hass, entity, entity_observation.get("to_state")
-            )
+            return condition.state(self.hass, entity, entity_observation.to_state)
         except ConditionError:
             return None
 
     def _process_multi_state(self, entity_observation):
-        """Return True if state conditions are met."""
-        entity = entity_observation["entity_id"]
+        """Return True if state conditions are met, otherwise return None.
+
+        Never return False as all other states should have their own probabilities configured.
+        """
+
+        entity = entity_observation.entity_id
 
         try:
-            if condition.state(self.hass, entity, entity_observation.get("to_state")):
+            if condition.state(self.hass, entity, entity_observation.to_state):
                 return True
         except ConditionError:
             return None
+        return None
 
     @property
     def extra_state_attributes(self):
         """Return the state attributes of the sensor."""
         attr_observations_list = [
-            obs.copy() for obs in self.current_observations.values() if obs is not None
+            observation.to_dict()
+            for observation in self.current_observations.values()
+            if observation is not None
         ]
 
-        for item in attr_observations_list:
-            item.pop("value_template", None)
-
         return {
-            ATTR_OBSERVATIONS: attr_observations_list,
+            ATTR_PROBABILITY: round(self.probability, 2),
+            ATTR_PROBABILITY_THRESHOLD: self._probability_threshold,
             ATTR_OCCURRED_OBSERVATION_ENTITIES: list(
                 {
-                    obs.get("entity_id")
-                    for obs in self.current_observations.values()
-                    if obs is not None
-                    and obs.get("entity_id") is not None
-                    and obs.get("observation") is not None
+                    observation.entity_id
+                    for observation in self.current_observations.values()
+                    if observation is not None
+                    and observation.entity_id is not None
+                    and observation.observed is not None
                 }
             ),
-            ATTR_PROBABILITY: round(self.probability, 2),
-            ATTR_PROBABILITY_THRESHOLD: self._probability_threshold,
+            ATTR_OBSERVATIONS: attr_observations_list,
```

Comment:
I think we don't need to use an intermediate variable here.
```suggestion
            ATTR_OBSERVATIONS: [
                observation.to_dict()
                for observation in self.current_observations.values()
                if observation is not None
            ],
```